### PR TITLE
minor opt: minor optimization to the orca parser

### PR DESCRIPTION
**Pull request by wbpcode**
Commit Message: minor opt: minor optimization to the orca parser
Additional Description:

By this way, the parser needn't to scan the whole header value if the header value has invalid format. And the we needn't create a copy of the header value for json format now.

Risk Level: low.
Testing: n/a.
Docs Changes: n/a.
Release Notes: n/a.
Platform Specific Features: n/a.

### CODE REVIEW

File: source/common/orca/orca_parser.cc
Comment (adisuissa) [inline source/common/orca/orca_parser.cc:176]: I'd consider keeping some string as part of the error.
I think it makes sense to keep the minimum between the first 5 characters, and the prefix up to the first occurrence of a " " (the delimiter).
It makes it easier to debug if/when things go wrong with an external upstream server.

**Review by adisuissa — COMMENTED** [review]
Thanks!
Overall LGTM, modulo minor comment.
Assigning Blake as the original contributor of this code.
/assign @blake-snyder

**Comment by alyssawilk** [comment]
(yeah I just invited blake to the org but I think it hasn't gone through yet =P)

File: source/common/orca/orca_parser.cc
Comment (wbpcode) [reply source/common/orca/orca_parser.cc:176]: SGTM.

File: source/common/protobuf/utility.h
Comment (blake-snyder) [inline source/common/protobuf/utility.h:1]: Does it make sense to split changes to the utility files/methods out into another PR as their impact isn't limited to ORCA code?

File: test/common/orca/orca_parser_test.cc
Comment (blake-snyder) [inline test/common/orca/orca_parser_test.cc:21]: nit: these can be removed now.

File: source/common/protobuf/yaml_utility.cc
Comment (blake-snyder) [inline source/common/protobuf/yaml_utility.cc:1]: Does it make sense to split changes to the utility files/methods out into another PR as their impact isn't limited to ORCA code?

File: source/common/protobuf/utility.h
Comment (wbpcode) [reply source/common/protobuf/utility.h:1]: OKay to me.

File: test/common/orca/orca_parser_test.cc
Comment (wbpcode) [reply test/common/orca/orca_parser_test.cc:21]: sgtm.

**Comment by wbpcode** [comment]
block by https://github.com/envoyproxy/envoy/pull/36525

File: source/common/protobuf/yaml_utility.cc
Comment (wbpcode) [reply source/common/protobuf/yaml_utility.cc:1]: See https://github.com/envoyproxy/envoy/pull/36525

**Comment by wbpcode** [comment]
gently ping @blake-snyder cc @alyssawilk

**Review by blake-snyder — APPROVED** [review]

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!

**Comment by wbpcode** [comment]
cc @adisuissa @blake-snyder thanks for the approvals 🌷

**Comment by wbpcode** [comment]
cc @alyssawilk friendly ping for an approval from senior maintainer because this change the core code. :)

**Review by alyssawilk — APPROVED** [review]
LGTM but general question is why is this in core espeically if it's not being used? May be a question for @blake-snyder

**Comment by wbpcode** [comment]
> LGTM but general question is why is this in core espeically if it's not being used? May be a question for @blake-snyder 

I think this should has been used by the orca related features?

**Comment by blake-snyder** [comment]
> LGTM but general question is why is this in core espeically if it's not being used? May be a question for @blake-snyder

This is currently used during ORCA header processing: https://github.com/envoyproxy/envoy/blob/main/source/common/router/router.cc#L2131

**Pull request merged**
